art/step/steps.py

```diff
@@ -42,9 +43,11 @@ def do(self, previous_states: Dict):
         Args:
             previous_states (Dict): previous states
         """
-        self.model.ml_train({"dataloader": self.datamodule.train_dataloader()})
-        self.validate(trainer_kwargs={"datamodule": self.datamodule})
-
+        model = self.model_class()
+        model.ml_train({"dataloader": self.datamodule.train_dataloader()})
+        model.set_metric_calculator(self.metric_calculator)
+        result = self.trainer.validate(model=model, datamodule= self.datamodule)
+        self.results["scores"].update(result[0])
```

Comment:
I wonder if we could move ml_train to step, then those lines would be handled by default inside the initialize_model.

Reply:
Since ML train will be used just by EvaluateBaseline step I would rather keep it here